Yosemite/YosemiteTests/Stores/NotificationCountStoreTests.swift

```diff
@@ -0,0 +1,166 @@
+import XCTest
+
+@testable import Storage
+@testable import Yosemite
+
+final class NotificationCountStoreTests: XCTestCase {
+    /// Mockup Dispatcher!
+    ///
+    private var dispatcher: Dispatcher!
+
+    /// Mockup Storage: InMemory
+    ///
+    private var storageManager: MockupStorageManager!
+
+    /// Mockup File Storage: Load data in memory
+    ///
+    private var fileStorage: MockInMemoryStorage!
+
+    /// Test subject
+    ///
+    private var subject: NotificationCountStore!
+
+    private let defaultSiteID: Int64 = 134
+
+    override func setUp() {
+        super.setUp()
+        dispatcher = Dispatcher()
+        storageManager = MockupStorageManager()
+        fileStorage = MockInMemoryStorage()
+        subject = NotificationCountStore(dispatcher: dispatcher, storageManager: storageManager, fileStorage: fileStorage)
+    }
+
+    override func tearDown() {
+        dispatcher = nil
+        storageManager = nil
+        fileStorage = nil
+        subject = nil
+        super.tearDown()
+    }
+
+    // MARK: `incrementNotificationCount`
+
+    func testIncrementingAndLoadingNotificationCountReturnsTheCorrectCount() {
+        let data = NotificationCountPListWrapper(notificationCountBySite: [defaultSiteID: [.comment: 2, .storeOrder: 6]])
+        try! fileStorage.write(data, to: URL(fileURLWithPath: ""))
+
+        var notificationCount: Int?
+        waitForExpectation { expectation in
+            let action = NotificationCountAction.incrementNotificationCount(siteID: defaultSiteID, type: .comment, incrementCount: 10) {
+                let loadAction = NotificationCountAction.loadNotificationCount(siteID: self.defaultSiteID, type: .comment) { count in
+                    notificationCount = count
+                    expectation.fulfill()
+                }
+                self.subject.dispatcher.dispatch(loadAction)
```

Comment:
I'm curious why the `self.dispatcher` is not enough. The test still seems to pass. 🤔 Or maybe `subject.onAction`?

Reply:
no particular reason, just from the other tests in the codebase 😅 I like `subject.onAction` so that it doesn't depend on how the dispatcher works 👍 updated in https://github.com/woocommerce/woocommerce-ios/pull/2320/commits/df4256ff84a5821d71b53e43c44c217f4696eb67